tests/unit-tests/graphpartitioning/test_metadata.py

```diff
@@ -0,0 +1,731 @@
+@pytest.mark.parametrize("edge_fmt_del", [" ", "/"])
+@pytest.mark.parametrize("create_files", [True, False])
+@pytest.mark.parametrize("prefix_dir", ["", "edges_"])
+@pytest.mark.parametrize("is_absolute", [True, False])
+def test_etype_files(
+    etypes, edge_fmt, edge_fmt_del, create_files, prefix_dir, is_absolute
+):
+    """Helper function to test various scenarions with edge files.
+    Files presence/absence is test rigorously.
+    """
+    input_dict = {}
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        # Add nodes to the metadata.
+        add_nodes(input_dict, {ntype: 10 for ntype in ntypes})
+
+        file_metadata = namedtuple("file_metadata", "")
+        file_metadata.root_dir = root_dir
+        file_metadata.prefix_dir = prefix_dir
+        file_metadata.create_files = create_files
+        file_metadata.is_absolute = is_absolute
+        file_metadata.file_prefix = "edges_"
+
+        # Add edges and create edge files, if necessary
+        add_edges_files(
+            input_dict,
+            {etype: 10 for etype in etypes},
+            {etype: idx + 1 for idx, etype in enumerate(etypes)},
+            {"name": edge_fmt, "delimiter": edge_fmt_del},
+            file_metadata,
+        )
+
+        # Create metadata.json file.
+        filename = os.path.join(root_dir, METADATA_NAME)
+        with open(filename, "w") as input_handle:
+            json.dump(input_dict, input_handle, indent=4)
+
+        # Test case here.
+        obj = MetadataSchema(METADATA_NAME, root_dir, init_maps=False)
+        obj._init_etype_maps()
+
+        # Check the initialization
+        try:
+            obj._init_etype_files()
+
+            if create_files:
+                for idx, etype in enumerate(etypes):
+                    (file_type, delimiter, data_files) = obj.get_etype_info(
+                        etype
+                    )
+                    assert file_type == edge_fmt, f"File type does not match."
+                    assert (
+                        delimiter == edge_fmt_del
+                    ), f"Delimiters does not match."
+                    assert (
+                        len(data_files) == idx + 1
+                    ), f"No. of edge files does not match."
+
+        except AssertionError as exp:
+            if create_files:
+                raise exp
+            else:
+                pass
+
+
+@pytest.mark.parametrize("ntypes", [[], [NODE1], [NODE1, NODE2]])
+@pytest.mark.parametrize("file_type", ["parquet", "numpy"])
+@pytest.mark.parametrize("delimiter", [" ", "/"])
+@pytest.mark.parametrize("create_files", [True, False])
+@pytest.mark.parametrize("prefix_dir", ["", "node_data"])
+@pytest.mark.parametrize("is_absolute", [True, False])
+def test_node_features(
+    ntypes, file_type, delimiter, create_files, prefix_dir, is_absolute
+):
+    """Unit tests when node features are present in the input graph."""
+
+    def run_test(root_dir, input_dict):
```

Comment:
this is a very large function. can you define the function outside the test function?

Reply:
This file has been re-organized from scratch... mostly to address @thvasilo comments. Now this is outdated.